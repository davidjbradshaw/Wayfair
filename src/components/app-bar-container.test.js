import { cleanup, fireEvent, render } from '@testing-library/react'
import { identity } from 'ramda'

import AppBarContanier from './app-bar-container'

const testAppBar = (open, setOpen = identity) =>
  render(<AppBarContanier open={open} setOpen={setOpen} />)

const snapContainer = (open) =>
  expect(testAppBar(open).container).toMatchSnapshot()

describe('AppBarContanier', () => {
  afterEach(() => {
    cleanup()
  })

  // eslint-disable-next-line jest/expect-expect
  test('renders (open)', () => {
    snapContainer(true)
  })

  // eslint-disable-next-line jest/expect-expect
  test('renders (closed)', () => {
    snapContainer(false)
  })

  test('setOpen called when clicked', () => {
    const setOpen = jest.fn(identity)
    const burgerButton = testAppBar(false, setOpen).getByTestId('sidebar-open')

    expect(setOpen).toHaveBeenCalledTimes(0)
    fireEvent.click(burgerButton)
    expect(setOpen).toHaveBeenCalled()
  })
})
