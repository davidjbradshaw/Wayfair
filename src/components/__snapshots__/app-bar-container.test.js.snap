// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarContanier renders (closed) 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        data-testid="sidebar-open"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <header
        style="color: rgb(127, 24, 127);"
      >
        <img
          alt="Wayfair - Partner Home"
          src="logo.svg"
          style="height: 28px; pointer-events: none; margin: 15px 20px 10px;"
        />
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          style="position: absolute; top: 14px; right: 0px; color: rgb(127, 24, 127);"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="material-icons md-24"
            >
              account_circle
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </header>
    </div>
  </header>
</div>
`;

exports[`AppBarContanier renders (open) 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 makeStyles-appBarShift-2 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-3 makeStyles-hide-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        data-testid="sidebar-open"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <header
        style="color: rgb(127, 24, 127);"
      >
        <img
          alt="Wayfair - Partner Home"
          src="logo.svg"
          style="height: 28px; pointer-events: none; margin: 15px 20px 10px;"
        />
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          style="position: absolute; top: 14px; right: 0px; color: rgb(127, 24, 127);"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="material-icons md-24"
            >
              account_circle
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </header>
    </div>
  </header>
</div>
`;
